rclcpp/src/rclcpp/executor.cpp

```diff
@@ -225,18 +224,17 @@ Executor::add_callback_group_to_map(
   }
   // Also add to the map that contains all callback groups
   weak_groups_to_nodes_.insert(std::make_pair(weak_group_ptr, node_ptr));
-  if (is_new_node) {
-    rclcpp::node_interfaces::NodeBaseInterface::WeakPtr node_weak_ptr(node_ptr);
-    weak_nodes_to_guard_conditions_[node_weak_ptr] = node_ptr->get_notify_guard_condition();
-    if (notify) {
-      // Interrupt waiting to handle new node
-      rcl_ret_t ret = rcl_trigger_guard_condition(&interrupt_guard_condition_);
-      if (ret != RCL_RET_OK) {
-        throw_from_rcl_error(ret, "Failed to trigger guard condition on callback group add");
-      }
+  weak_groups_to_guard_conditions_[weak_group_ptr] =
```

Comment:
just to confirm, this does not need to check if `weak_group_ptr` is already in the map, since it already checks with `get_associated_with_executor_atomic` above?

Reply:
Yes